src/string/strlen.rs

```diff
@@ -3,12 +3,44 @@ use core::usize;
 use c_types::*;
 
 #[no_mangle]
-pub unsafe extern "C" fn strlen(s: *const c_schar) -> size_t {
-    // TODO(adam) convert to checking word-size chunks
-    for i in 0.. {
-        if *s.offset(i) == 0 {
-            return i as usize;
+pub unsafe extern "C" fn strlen(mut s: *const c_schar) -> size_t {
+    
+    #[inline]
+    fn align() -> usize { core::mem::size_of::<size_t>() }
```

Comment:
Because this isn't a root module, the `core` namespace isn't automatically available, and this won't build. Changing it to this works fine:

```rust
#[inline]
fn align() -> usize {
    use core::mem;
    mem::size_of::<size_t>()
}
```